pkg/security/secl/compiler/generators/accessors/accessors.go

```diff
@@ -119,26 +119,37 @@ func handleField(module *common.Module, astFile *ast.File, name, alias, prefix,
 			aliasPrefix = alias
 		}
 
-		spec := astFile.Scope.Lookup(fieldType.Name)
+		spec := astFile.Scope.Lookup(fieldType)
 		handleSpec(module, astFile, spec.Decl, prefix, aliasPrefix, event, iterator, dejavu)
 	}
 
 	return nil
 }
 
-func getFieldIdent(field *ast.Field) (ident *ast.Ident, isPointer, isArray bool) {
-	if fieldType, ok := field.Type.(*ast.Ident); ok {
-		return fieldType, false, false
-	} else if fieldType, ok := field.Type.(*ast.StarExpr); ok {
-		if ident, ok := fieldType.X.(*ast.Ident); ok {
-			return ident, true, false
-		}
-	} else if ft, ok := field.Type.(*ast.ArrayType); ok {
-		if ident, ok := ft.Elt.(*ast.Ident); ok {
-			return ident, false, true
-		}
+func getFieldName(expr ast.Expr) string {
+	switch expr.(type) {
```

Comment:
nit: switch expr := expr.(type)
to avoid all the casts

Reply:
Nice ! I forgot this was possible :)